### ISSUE TITLE
How can I make connection automatically?

### ISSUE THREAD

**Issue by jinserk** [NoahZinsmeister/web3-react#12]
Hi, @NoahZinsmeister again!

I made my first react-app successfully connected when I login MetaMask. now my app has the codes:
App.js:
```
import React, { Component } from "react";
import Web3Provider, { Connectors, useWeb3Context } from "web3-react";

import './App.css';
import { Web3Unavailable, AccountUnavailable } from "./Error";

const { MetaMaskConnector } = Connectors;
const MetaMask = new MetaMaskConnector();
const connectors = { MetaMask };

function MyComponent() {
  const context = useWeb3Context();
  const ether = context.library;

  React.useEffect(() => {
    context.setConnector("MetaMask")
  }, []);
  console.log(context);

  return (
    <React.Fragment>
      {context.active ? (
        <React.Fragment>
          <p>Active Connector: {context.connectorName || "None"}</p>
          <p>Account: {context.account || "None"}</p>
          <p>Network ID: {context.networkId || "None"}</p>
          <p>test: {ether.s} </p>
        </React.Fragment>
      ) : (
        <React.Fragment>
          <Web3Unavailable />
        </React.Fragment>
      )}
      {context.error && <p>{context.error.toString()}</p>}
    </React.Fragment>
  );
}

class App extends Component {
  render() {
    return (
      <Web3Provider connectors={connectors} libraryName="ethers.js">
        <div className="App">
          <MyComponent />
        </div>
      </Web3Provider>
    );
  }
}

export default App;
```

So I plan to use only MetaMaskConnector and handle all connection using it. If the MetaMask is not installed or not logged in, it shows some Error page. 
Most of the components are working as I expected, but on the error page, it doesn't connect automatically when I logged in MetaMask using its plugin window.
Can you advise me how to do so?

One more thing is, how can I access the ether's current provider? I want to make some button to make transfer or something. But I'm not so familiar with the latest javascript and to deal with the context as well.

Thank you!

**Comment by NoahZinsmeister** [comment]
Nothing about this looks obviously wrong to me, you're saying when you load this page the context remains inactive? Also, if you're using ethers.js, `context.library` will be an [ethers provider](https://docs.ethers.io/ethers.js/html/api-providers.html) that you can use to do transaction sending/etc.

**Comment by jinserk** [comment]
What I want to do is that once the page has been loaded but there is no connection, then it shows error page and unchanged even though I log in the MetaMask. I'd like to know how to reload the page if there is an available connection.

**Comment by jinserk** [comment]
I'm still not sure that `context.library` is identical to the current web3 provider, since it shows only "None" if I added this:
```
  const provider = context.library;
  return(
    <React.Fragment>
      <p>test: {provider.blockNumber || "None"}</p> 
    </React.Fragment>
  );
```
so, how can I check if I get the correct web3 provider?

**Comment by NoahZinsmeister** [comment]
@jinserk if you set up a codesandbox with your (broken) example, I will take a look! For what it's worth, the `.blockNumber` property will be undefined even for a properly initialized connector, you have to await `.getBlockNumber()`. But, if `context.library` is undefined, that means the connector has not been set.

**Comment by NoahZinsmeister** [comment]
closing this for now, let me know when you have a reproducible example and ill open it back up!

**Comment by jinserk** [comment]
@NoahZinsmeister, Thanks for consideration. I have a project repo on https://github.com/jinserk/web3-react-dapp for open review, but I don't know how to upload this into codesandbox. I'll try to do it today. Thank you again!

**Comment by NoahZinsmeister** [comment]
hey! i took a look at your project, cleaned some things up, and it works now! see [my PR](https://github.com/jinserk/web3-react-dapp/pull/1).

**Comment by jinserk** [comment]
Oh, many thanks for this!! Since I'm participating in a conference till
Wed, I'm not so free to use computer now. Will check after back to office.
Thank you so much again!

**Comment by jinserk** [comment]
I confirmed that your PR resolves the re-rendering issue of the latter MetaMask login. Thanks again for that and for letting me know about the new update version!

About the use of context.library, I am still looking for the solution how to properly render the response of Promises from the ethers or web3. Using the `useEffect` is not the correct solution since I cannot use the callback of `.then` in `return()` of the react's function component. What I found is something to use `setState` in the react's class component, but in this case I cannot use `useWeb3Context()` hook. Can I ask you some advice for this over `web3-react`?
